src/main/php/web/auth/oauth/OAuth2Backend.class.php

```diff
@@ -0,0 +1,89 @@
+<?php namespace web\auth\oauth;
+
+use peer\http\HttpConnection;
+use lang\IllegalStateException;
+use io\streams\Streams;
+
+class OAuth2Backend {
```

Comment:
Is this the best name we can come up with? How is this called in other OAuth2 APIs?

Reply:
Here https://oauth2-client.thephpleague.com/usage/, they call this *Provider*

Reply:
GPT additionally suggests:

**Consumer**: This term is often used in the context of APIs, where the consumer is the entity (application, device, or software) that consumes or interacts with the server's provided services.

**Endpoint**: In the context of APIs, an endpoint refers to a specific URL or URI that the client interacts with to access the server's functionalities and resources.

Reply:
After putting some though into this I prefer to use *Endpoint* instead of *Backend*, as we are not instantiating a backend here.